app/main.py

```diff
@@ -1,2 +1,48 @@
+from typing import Any, Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+        while (self.hash_table[index] is not None
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+        return index
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.__init__(self.capacity * 2)
+        for item in old_hash_table:
+            if item is not None:
+                self.__setitem__(item.key, item.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            if self.size + 1 >= self.capacity * 2 / 3:
+                self.resize()
+                return self.__setitem__(key, value)
```

Comment:
can you rewrite this without this recursive call?  you can simply check if table needs resizing, resize if necessary, then calculate index and set node..

Reply:
not done, if you don't need a recursive call why would you use it? recursion can increase complexity of code, need to use it carefully